provider/datagen/src/transform/cldr/plurals/mod.rs

```diff
@@ -74,7 +74,7 @@ impl<M: ResourceMarker<Yokeable = PluralRulesV1<'static>>> IterableResourceProvi
         Ok(self
             .get_rules_for(M::KEY)?
             .0
-            .iter_keys()
+            .keys()
```

Comment:
Issue, here and elsewhere: one of the motivations for migrating from HashMap to LiteMap (#1341) was that it gives us stable ordering when calling `.iter()`, which is a desirable property. When generating data, you _really_ don't want to introduce entropy based on what state a hash map is in.

Reply:
The output of `supported_options` is `par_iter`ed, so there's no point whatsoever in this being stable.

There was only one case where generating code expected a serde map to be sorted and I used a `BTreeMap` for that.

Reply:
ok; I'm a little worried that there are `.iter()` calls elsewhere in datagen that assumed that things were sorted. You found one of them, presumably because there was a test/diff that caught it, but how confident are we that there aren't more? To me this is a problem that should be handled by having separate method names, like `.iter_sorted()` that LiteMap and BTreeMap implement but not HashMap. That said, this is only a theoretical concern short of double-checking all of the code in datagen.

Reply:
I actually double checked most of the code, but not all (newtype structs are annoying because you can search for field names).